src/System/Components/RouterLink.tsx

```diff
@@ -84,11 +88,31 @@ export const RouterLink: React.FC<React.PropsWithChildren<React.PropsWithChildre
       }
     }
 
+    // Inject a query param to indicate that the link was successfully pre-fetched.
+    const updatedHrefWithPrefetchQueryParam = (
+      to: string,
+      isPrefetched: boolean
+    ): string => {
+      const [baseUrl, queryString] = to.split("?")
+      const queryParams = new URLSearchParams(queryString || "")
+      if (isPrefetched) {
+        queryParams.set("is_prefetched", "true") // intentional underscore casing
+      }
+      return `${baseUrl}?${queryParams.toString()}`
+    }
+
+    const isArtistRoute = to?.startsWith("/artist/")
+
     if (isRouterAware) {
       return (
         <RouterAwareLink
           inline={inline}
-          to={to ?? ""}
+          // Update the href to include a query param if the link was a pre-fetched artist route.
+          to={
+            (isArtistRoute
```

Comment:
we don't want `?is_prefetched=true` showing up on every pre-fetched link all over the place, so just adding it to artist routes (can probably be improved).

I don't love squatting on a query param for this, but this feels fine. Should we remove the query param from the location via middleware? That way people won't copy-paste that version of the URL.